website/docs/reference/global-configs/behavior-changes.md

```diff
@@ -269,6 +271,35 @@ When you set the `validate_macro_args` flag to `True`, dbt will:
 - If no arguments are documented in the YAML, infer them from the macro and include them in the [`manifest.json` file](/reference/artifacts/manifest-json)
 
 
+### Warn-error handler for all warnings
+
+By default, the `require_all_warnings_handled_by_warn_error` flag is set to `False`.
+
+When you set `require_all_warnings_handled_by_warn_error` to `True`, all warnings raised during a run are routed through the `--warn-error` / `--warn-error-options` handler. This ensures consistent behavior when promoting warnings to errors or silencing them. When the flag is `False`, only some warnings are processed by the handler while others may bypass it.
+
+Note that enabling this for projects that use `--warn-error` (or `--warn-error-options='{"error":"all"}'`) may cause builds to fail on warnings that were previously ignored. We recommend enabling it gradually.
+
+<Expandable alt_header="Recommended steps to enable the flag">
+
+When the `require_all_warnings_handled_by_warn_error` flag is set to `True`, we recommend the following rollout plan:
+
+1. Run a full build without partial parsing to surface parse-time warnings, and confirm it finishes successfully:
+
+   ```bash
+   dbt build --no-partial-parse
+   ```
+
+   - Some warnings are only emitted at parse time.
+   - If the build fails because warnings are already treated as errors (via `--warn-error` or `--warn-error-options`), fix those first and re-run.
+2. Review the logs:
+   - If you see warnings, you have warnings that aren’t currently handled by `--warn-error`/`--warn-error-options`. Continue to the next step.
```

Comment:
@QMalcolm i'm not sure i understand how you'd know these warning aren't currently handled by `--warn-error`?

is this _just in the case_ that you're using `--warn-error` (or `--warn-error-options='{"error":"all"}'`)